core/src/sql/cast.rs

```diff
@@ -42,7 +42,10 @@ impl Cast {
 		doc: Option<&CursorDoc<'_>>,
 	) -> Result<Value, Error> {
 		// Compute the value to be cast and convert it
-		stk.run(|stk| self.1.compute(stk, ctx, opt, doc)).await?.convert_to(&self.0)
+		match &self.0 {
+			Kind::Closure => self.1.clone().convert_to(&Kind::Closure),
```

Comment:
Can we add a comment here to explain why we are not computing the value before casting it?

Reply:
```suggestion
            // We do not compute the value here. As an example, 
            // casting a block into a closure converts it in a way 
            // where it stays uncomputed, until the closure is executed
			Kind::Closure => self.1.clone().convert_to(&Kind::Closure),
```